### PR TITLE
Feature.ab deployment.alt backend pools

### PR DESCRIPTION
**Pull request by russokj**
WIP - Feature.ab_deployment branch is messed up, but the commit 'Added creation of pools for alternatedBackends' is at least reviewable (as well as the follow on commit based on review comments).

### CODE REVIEW

File: pkg/appmanager/appManager.go [hunk 1]
```diff
@@ -1457,6 +1471,7 @@ func (appMgr *Manager) handleConfigForType(
 
 	deactivated := false
 	if _, ok := svcPortMap[pool.ServicePort]; !ok {
+			pool.ServiceName, svcKey.Namespace)
```
Comment (sjberman) [inline pkg/appmanager/appManager.go]: This feels out of place.

File: pkg/appmanager/resourceConfig.go [hunk 2]
```diff
@@ -980,6 +1016,7 @@ func createRSConfigFromRoute(
 		}
 		if !found {
 			rsCfg.Pools = append(rsCfg.Pools, pool)
+		} else {
```
Comment (sjberman) [inline pkg/appmanager/resourceConfig.go]: Empty else here.

File: pkg/appmanager/routing.go [hunk 3]
```diff
@@ -394,7 +394,7 @@ func updateDataGroupForPassthroughRoute(
 	dgMap InternalDataGroupMap,
 ) {
 	hostName := route.Spec.Host
-	poolName := formatRoutePoolName(route)
+	poolName := formatRoutePoolName(route, route.Spec.To.Name) // TODO(kenr): Do we need to make this svcName? (same below)
```
Comment (sjberman) [inline pkg/appmanager/routing.go]: I think we would want this to be svcName. If a Route is going to specify more than one backend, then we want the passthrough/reencrypt rules to apply to any pools for that route. So the data group would need to know about each pool for the Route.

Comment (sjberman) [reply pkg/appmanager/routing.go]: It may be worth getting @f5gary's input on this as well.

File: pkg/appmanager/appManager.go (hunk 1 above)
Comment (russokj) [reply pkg/appmanager/appManager.go]: It is.  It was part of a print statement I forgot to completely remove.

File: pkg/appmanager/appManager_test.go [hunk 4]
```diff
@@ -3179,11 +3179,12 @@ var _ = Describe("AppManager Tests", func() {
 					}
 					hostDg, found := mockMgr.appMgr.intDgMap[hostDgKey]
 					Expect(found).To(BeTrue())
+					// TODO(kenr): READD AFTER UNDERSTANDING WHY THESE FAILED
```
Comment (russokj) [inline pkg/appmanager/appManager_test.go]: This comment is no longer needed.  The tests no longer fail.

**Comment by sjberman** [comment]
Make sure you rebase on master, as the recent Shared Ingress work could have some impact on how this code will run (specifically around cleanup). There are also some conflicts.

**Comment by russokj** [comment]
Closing out this review since the rebase wasn't properly done.  A new review is out that is against the feature branch which was updated with the latest from master (https://github.com/F5Networks/k8s-bigip-ctlr/pull/475).  That request should include the latest code changes.

**Pull request closed without merging**